### PR TITLE
{Storage} `az storage share-rm`: Snapshots related operations

### PR DESCRIPTION
**Pull request by Juliehzl**
**Description**

- BREAKING CHANGE: Raise error when there are snapshots for target file share 
- Support listing with deleted shares and their snapshots

**Testing Guide**

**History Notes**

[Storage] BREAKING CHANGE: `az storage share-rm delete`: Raise error when there are snapshots for target file share and add `--include` to specify deleting target file share and its snapshots
[Storage] `az storage share-rm list`: Support list with snapshots.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Storage

File: src/azure-cli/azure/cli/command_modules/storage/_help.py [hunk 1]
```diff
@@ -2260,7 +2260,7 @@
 helps['storage share-rm delete'] = """
 type: command
 short-summary: Delete the specified Azure file share or share snapshot.
-long-summary: 'Incoming BREAKING CHANGE: Snapshot will not be deleted by default and we will add a new parameter to use if you want to include sanpshots for delete operation.'
+long-summary: 'BREAKING CHANGE: Snapshot will not be deleted by default and we will add a new parameter to use if you want to include sanpshots for delete operation.'
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/_help.py:2263]:
```suggestion
long-summary: 'BREAKING CHANGE: Snapshot can not be deleted by default and we have added a new parameter to use if you want to include sanpshots for delete operation.'
```

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 2]
```diff
@@ -1206,6 +1206,9 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
                    help='The DateTime value that specifies the share snapshot to retrieve.')
         c.ignore('filter', 'maxpagesize')
 
+    with self.argument_context('storage share-rm delete', resource_type=ResourceType.MGMT_STORAGE) as c:
+        c.argument('include', default='none')
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py:1210]: Why the default value is not `None` or `'None'`?

**Review by evelyn-ys — APPROVED** [review]
LGTM

**Pull request merged**